### DISCUSSION TITLE
TrackViewer v2 feedback

### DISCUSSION THREAD

**Discussion by perpetualKid** [perpetualKid/FreeTrainSimulator#49] [General]
Collecting ideas and feedback for further development and priorization of feature requests for new Trackviewer

![Route Overview](https://github.com/perpetualKid/ORTS-MG/wiki/media/TrackViewerPreview2.png)

**Comment by scottypitcher** [comment]
Thanks for the work you've done. My son and I wanted to run OR or MSTS with Wine or with QEMU for a long time. This ORMG is great!

The trackviewer tool is great and I can move around a route (using Marias Pass v5) much quicker than the old MSTS Activity Editor. My only comment is that it will be good to edit paths. I'm trying to set up some Timetables for the route, and I'm having to load MSTS Route Editor to do the path, because OR Trackviewer doesn't work and I don't think TSRE can do more than copy the paths (it doesn't seem to have any path edit function I can see in UI or the PDF).

Other than that - great. We've waited a few years to get OR running from the Linux desktop even via Wine, and we've really enjoyed running in these past few weeks, odd glitches and crashes aside.

Scott

**Reply by perpetualKid** [reply]
thanks for the feedback which is encouraging to continue efforts on new editor tools! 

Not sure which version you are using, it may be worth to switch to the latest development builds and use the buildin updater to get a recent version. The TrackViewer has been renamed to highlight the intent to become a moe universal Toolbox, and work has been started to bring Path editing capabilities as well:

![image](https://github.com/user-attachments/assets/a2670807-2c18-4c79-b22d-2899a1c8c6f7)

However, that work had been paused in favor of a complete refactoring of the data formats being used, to overcome the limitations of the legacy KUJU file formats used in MSTS and subsequently OpenRails. 
While this work is now progressing well, it will still take efforts to have these new formats enabled across the codebase, and eventually been used when resuming the work on path editor.

**Comment by perpetualKid** [comment]
@scottypitcher the toolbox in Development version, now has a working version of path editor!
It's early stages, and some work to do, maybe give it a try and share feedback!

**Reply by scottypitcher** [reply]
Ah cool. Thanks for the heads up. I haven't built from source yet so I'll setup a dev system on Win7 VM and give it a go.

**Reply by perpetualKid** [reply]
the development builds are also available as binaries, through the "Developer Mode" channel in update settings, no need to compile a version yourself

![image](https://github.com/user-attachments/assets/6a914a9b-37cc-4654-897d-42e9334345df)
